File: programmers/level1/로또의_최고_순위와_최저_순위/Solution.js
function solution(lottos, win_nums) {
    var answer = [];

    let match = 0;
    let zeros = 0;

    lottos.forEach(lotto => {
        if (lotto === 0) {
            zeros++;
        }
    });

    win_nums.forEach(win_num => {
        match += lottos.filter(lotto => win_num === lotto).length;
    });

    answer[0] = 7 - (match + zeros) < 7 ? 7 - (match + zeros) : 6;
    answer[1] = 7 - match < 7 ? 7 - match : 6;

    return answer;
}

function init() {
    lottos = [44, 1, 0, 0, 31, 25];
    win_nums = [31, 10, 45, 1, 6, 19];

    console.log(solution(lottos, win_nums));
}

init();

File: programmers/level1/로또의_최고_순위와_최저_순위/solution.py
def solution(lottos, win_nums):
    answer = [0, 0]

    match = 0
    zeros = lottos.count(0)

    for win_num in win_nums:
        if win_num in lottos:
            match += 1

    answer[0] = 7 - (match + zeros) if 7 - (match + zeros) < 7 else 6
    answer[1] = 7 - match if 7 - match < 7 else 6

    return answer


if __name__ == "__main__":
    lottos = [44, 1, 0, 0, 31, 25]
    win_nums = [31, 10, 45, 1, 6, 19]

    print(solution(lottos, win_nums))
